File: GridView/RowDetailsInGridView/RowDetailsInGridView/RowDetailsRowElement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Telerik.WinControls.UI;

namespace RowDetailsInGridView
{
    public class RowDetailsRowElement : GridDataRowElement
    {
        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(GridDataRowElement);
            }
        }

        public override void UpdateInfo()
        {
            if (!this.RowInfo.IsCurrent)
            {
                this.RowInfo.Height = TableElement.RowHeight;
            }
            else
            {
                this.RowInfo.Height = ((RowDetailsGrid)this.GridControl).DetailsRowHeight;
            }
            base.UpdateInfo();
        }
    }
}

File: GridView/RowDetailsInGridView/RowDetailsInGridViewVB/RowDetailsRowElement.vb
Imports Telerik.WinControls.UI

Public Class RowDetailsRowElement
    Inherits GridDataRowElement
    Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
        Get
            Return GetType(GridDataRowElement)
        End Get
    End Property

    Public Overrides Sub UpdateInfo()
        If Not Me.RowInfo.IsCurrent Then
            Me.RowInfo.Height = TableElement.RowHeight
        Else
            Me.RowInfo.Height = (CType(Me.GridControl, RowDetailsGrid)).DetailsRowHeight
        End If
        MyBase.UpdateInfo()
    End Sub
End Class
